script/deploy/healthCheck.mts

```diff
@@ -0,0 +1,367 @@
+]
+
+const chainMap: Record<string, Chain> = {}
+for (const [k, v] of Object.entries(chains)) {
+  // @ts-ignore
+  chainMap[k] = v
+}
+
+const errors: string[] = []
+const main = defineCommand({
+  meta: {
+    name: 'LIFI Diamond Health Check',
+    description: 'Check that the diamond is configured correctly',
+  },
+  args: {
+    network: {
+      type: 'string',
+      description: 'EVM network to check',
+      required: true,
+    },
+  },
+  async run({ args }) {
+    if ((await $`${louperCmd}`.exitCode) !== 0) {
+      const answer = await consola.prompt(
+        'Louper CLI is required but not installed. Would you like to install it now?',
+        {
+          type: 'confirm',
+        }
+      )
+      if (answer) {
+        await spinner(
+          'Installing...',
+          () => $`npm install -g @mark3labs/louper-cli`
+        )
+      } else {
+        consola.error('Louper CLI is required to run this script')
+        process.exit(1)
+      }
+    }
+
+    const { network } = args
+    const deployedContracts = await import(
+      `../../deployments/${network.toLowerCase()}.json`
+    )
+    const dexs = (await import(`../../config/dexs.json`))[
+      network.toLowerCase()
+    ] as Address[]
+
+    const globalConfig = await import('../../config/global.json')
+
+    const chain = chainMap[network]
+    const publicClient = createPublicClient({
+      batch: { multicall: true },
+      chain,
+      transport: http(),
+    })
+
+    consola.info('Running post deployment checks...\n')
+
+    // Check core facets
+    consola.box('Checking Core Facets...')
+    for (const facet of coreFacets) {
+      if (!deployedContracts[facet]) {
+        logError(`Facet ${facet} not deployed`)
+        continue
+      }
+
+      consola.success(`Facet ${facet} deployed`)
+    }
+
+    // Checking Diamond Contract
+    consola.box('Checking diamond Contract...')
+    if (!deployedContracts['LiFiDiamond']) {
+      logError(`LiFiDiamond not deployed`)
+      finish()
+    } else {
+      consola.success('LiFiDiamond deployed')
+    }
+
+    const diamondAddress = deployedContracts['LiFiDiamond']
+
+    // Check that core facets are registered
+    consola.box('Checking facets registered in diamond...')
+    $.quiet = true
+    const facetsResult =
+      await $`${louperCmd} inspect diamond -a ${diamondAddress} -n ${network} --json`
+
+    const resgisteredFacets = JSON.parse(facetsResult.stdout).facets.map(
+      (f: { name: string }) => f.name
+    )
+
+    for (const facet of coreFacets) {
+      if (!resgisteredFacets.includes(facet)) {
+        logError(
+          `Facet ${facet} not registered in Diamond or possibly unverified`
+        )
+      } else {
+        consola.success(`Facet ${facet} registered in Diamond`)
+      }
+    }
+
+    // Check that core periphery facets are deployed
+    consola.box('Checking periphery contracts...')
+    for (const contract of corePeriphery) {
+      if (!deployedContracts[contract]) {
```

Comment:
I would suggest to check the addresses for deployed bytecode....just relying on the logs seems risky.
Same for the check if diamond is deployed above.

Reply:
did you see this comment @ezynda3 ?